tests/test_utils.py

```diff
@@ -306,3 +313,513 @@ def check_multistream_classic(fnames):
+
+    elif test_type == "science_test_fluxnet":
+        plot_fluxnet_comparison(driver,
+                                science_test_data_dir,
+                                compare_data,
+                                result_dir,
+                                plot_dir,
+                                plots_to_make)
+    else:
+        print("this has not been implemented in the VIC 5.0 science test suite")
+
+def plot_snotel_comparison(driver, science_test_data_dir,
+                            compare_data_dict,
+                            result_dir, plot_dir,
+                            plots_to_make):
+    ''' makes snotel figures
+
+    '''
+
+    # plot settings
+    plot_variables = ['OUT_SWE', 'OUT_ALBEDO', 'OUT_SALBEDO', 'OUT_SNOW_DEPTH',
+                    'OUT_SNOW_CANOPY', 'OUT_SNOW_PACK_TEMP', 'OUT_SNOW_MELT',
+                    'OUT_R_NET', 'OUT_LATENT', 'OUT_SENSIBLE']
+
+    plot_units = ['mm', 'fraction', 'fraction', 'mm', '%', 'degrees C',
+                    'mm', '$W/{m^2}$', '$W/{m^2}$', '$W/{m^2}$']
+    lw = 4.0
+    sns.set_style('whitegrid')
+
+    for filename in os.listdir(os.path.join(science_test_data_dir,
+                                            'datasets',
+                                            'snotel',
+                                            'observations')):
+
+        # get lat/lng from filename
+        file_split = re.split('_', filename)
+        lng = file_split[3].split('.txt')[0]
+        lat = file_split[2]
+
+        # loop over data to compare
+        data = {}
+        for key, items in compare_data_dict.items():
+
+            # read in data
+
+            if key == "snotel":
+
+                data[key] = read_snotel_swe_obs(filename,
+                                                science_test_data_dir,
+                                                items)
+
+            elif key == "VIC.4.2.d":
+
+                data[key] = read_vic_42_output(lat, lng,
+                                        science_test_data_dir,
+                                        items)
+
+            else:
+
+                data[key] = read_vic_5_output(lat, lng,
+                                        science_test_data_dir,
+                                        items)
+
+        # loop over variables to plot
+        for i, plot_variable in enumerate(plot_variables):
+
+            if 'water_year' in plots_to_make:
+
+                fig, ax = plt.subplots(figsize=(10,10))
+
+                df = pd.DataFrame({key: d[plot_variable] for key, d in
+                                            data.items() if plot_variable in d})
+
+                # merge DateTime indexes (time series lengths are different)
+                if 'snotel' in data.keys():
+                    df_join = data.pop('snotel').merge(next(iter(data.values())),
+                                                    how='outer', on='DATES')
+                    df.index = df_join['DATES']
```

Comment:
is `df_join` only used here? Shouldn't line 560 have taken care of the merging of the timeseries?